pkg/autodiscovery/providers/providers.go

```diff
@@ -50,3 +50,12 @@ type ConfigProvider interface {
 	// The result is displayed in diagnostic tools such as `agent status`.
 	GetConfigErrors() map[string]ErrorMsgSet
 }
+
+// StreamingConfigProvider is an interface used together with ConfigProvider.
+// ConfigProviders that are able to use streaming should implement it, and the
+// config poller will use Stream instead of Collect to collect config changes.
+type StreamingConfigProvider interface {
```

Comment:
Does it make sense to do a similar thing for CollectingConfigProvider, so that streaming providers don't have to implement Collect?

Reply:
Yeah, I think it does. `IsUpToDate` is also superfluous for the streaming providers now.